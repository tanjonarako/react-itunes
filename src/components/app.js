import React, { Component } from 'react';
import SearchBar from './search_bar';
import SearchResultList from './search_result_list';
import '../stylesheet/app.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <SearchBar/>
        <SearchResultList/>
      </div>
    );
  }
}

export default App;
