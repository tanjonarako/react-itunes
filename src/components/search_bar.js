import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const searchAction = (searchTerm) => ({
  type: "updateList",
  payload: {
    request: {
      url: 'https://itunes.apple.com/search?term=' + searchTerm
    }
  }
});

const mapDispatchToProps = (dispatch) => ({
  doSearch: (searchTerm) => dispatch(searchAction(searchTerm)),
});

export class SearchBar extends Component {
  state = {
    value: ''
  }

  static propTypes = {
    doSearch: PropTypes.func
  }

  handleSubmit = (e) => { 
    e && e.preventDefault();

    const inputValue = this.state.value;

    this.props.doSearch(inputValue);
    this.setState({value: ''});
  }

  handleChange = (e) => {
    this.setState({value: e.target.value});
  }

  render() {
    return (
      <header className="justify-content-center align-items-center search-bar">
        <div className="navbar bg-nav row-flex">
          <p className="navbar-brand m-0 py-0 pr-2" href="#">
            <img src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg" width="40" height="40" className="d-inline-block align-center" alt="itunes" />
            iTunes
          </p>
          <form className="search-form my-2 my-lg-0 col-flex" onSubmit={this.handleSubmit} >
            <input className="search-form__input bg-dark" type="search" placeholder="Search" value={this.state.value} onChange={this.handleChange} />
            <a className="search-form__button m-0 p-0" href="#" role="button" type="submit" onClick={this.handleSubmit} ><img src="https://icongr.am/clarity/search.svg?color=FFFFFF" width="20" alt="search icon"/></a>
          </form>
        </div>
      </header>
    );
  }
}

export default connect(
  state => ({}),
  mapDispatchToProps
)(SearchBar)
