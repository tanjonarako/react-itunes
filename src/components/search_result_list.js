import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchResultListItem from './search_result_list_item';


export class SearchResultList extends Component {
  static propTypes = {
    list: PropTypes.array
  }

  render() {
    const list = this.props.list;
    return (
      <div className="search-result-list">
        <ul className="list-unstyled">
          {list.map((item) => (
            <SearchResultListItem key={item.trackId} item={item}/>
          ))}
        </ul>
      </div>
    );
  }
}

SearchResultList.defaultProps = { list: []}

export default connect(
  state => ({
    list: state.list
  })
)(SearchResultList)
