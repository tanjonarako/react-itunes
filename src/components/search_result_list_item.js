import React from 'react';

const SearchResultListItem = (props) => (
  <li className="search-result-list__song row-flex p-3">
    <img src={props.item.artworkUrl60} width="60" height="60" alt="song logo"/>
    <div className="search-result-list__song-details pl-3 col-flex">
      <h2 className="search-result-list__song-name m-0">{props.item.trackName}</h2>
      <span className="search-result-list__song-artist m-0">{props.item.artistName}</span>
    </div>
  </li>
);

export default SearchResultListItem;
