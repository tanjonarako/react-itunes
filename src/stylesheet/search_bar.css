.bg-nav {
  background-color:#3e4a56;
  width:100%;
}

.search-bar {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  color: white;
}

.search-form__input, .search-form__input:focus{
  color:#fff;
  border: none;
  box-shadow: none;
  width: 100%;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  outline: none;
}

.search-form__button {
  position: absolute;
  right: 25px;
  top: 16px;
}
