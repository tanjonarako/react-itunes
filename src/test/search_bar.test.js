import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SearchBar } from '../components/search_bar';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchBar', () => {
  let wrapper

  beforeEach(()=>{
    wrapper = shallow(<SearchBar/>);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state on change of the input', () => {
    wrapper.find('input').simulate('change', 
      {target: { value: 'Ed Sheeran' }
    });

    expect(wrapper.state().value).toEqual('Ed Sheeran');
  });

  it('should call doSearch on submit', () => {
    const doSearch = jest.fn();
    const wrapper2 = shallow(<SearchBar doSearch={doSearch}/>);
    wrapper2.setState({ value: 'Ed Sheeran' });
    wrapper2.find('a').simulate('click');

    expect(doSearch).toBeCalledWith('Ed Sheeran');
  });
});
