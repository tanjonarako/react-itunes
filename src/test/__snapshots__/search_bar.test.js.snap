// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="navbar bg-nav row-flex"
>
        <p
                className="navbar-brand m-0 py-0 pr-2"
                href="#"
        >
                <img
                        alt="itunes"
                        className="d-inline-block align-center"
                        height="40"
                        src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                        width="40"
                />
                iTunes
        </p>
        <form
                className="search-form my-2 my-lg-0 col-flex"
                onSubmit={[Function]}
        >
                <input
                        className="search-form__input bg-dark"
                        onChange={[Function]}
                        placeholder="Search"
                        type="search"
                        value=""
                />
                <a
                        className="search-form__button m-0 p-0"
                        href="#"
                        onClick={[Function]}
                        role="button"
                        type="submit"
                >
                        <img
                                alt="search icon"
                                src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                width="20"
                        />
                </a>
        </form>
</div>,
      "className": "justify-content-center align-items-center search-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="navbar-brand m-0 py-0 pr-2"
            href="#"
>
            <img
                        alt="itunes"
                        className="d-inline-block align-center"
                        height="40"
                        src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                        width="40"
            />
            iTunes
</p>,
          <form
            className="search-form my-2 my-lg-0 col-flex"
            onSubmit={[Function]}
>
            <input
                        className="search-form__input bg-dark"
                        onChange={[Function]}
                        placeholder="Search"
                        type="search"
                        value=""
            />
            <a
                        className="search-form__button m-0 p-0"
                        href="#"
                        onClick={[Function]}
                        role="button"
                        type="submit"
            >
                        <img
                                    alt="search icon"
                                    src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                    width="20"
                        />
            </a>
</form>,
        ],
        "className": "navbar bg-nav row-flex",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="itunes"
                className="d-inline-block align-center"
                height="40"
                src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                width="40"
/>,
              "iTunes",
            ],
            "className": "navbar-brand m-0 py-0 pr-2",
            "href": "#",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "itunes",
                "className": "d-inline-block align-center",
                "height": "40",
                "src": "https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "iTunes",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search-form__input bg-dark"
                onChange={[Function]}
                placeholder="Search"
                type="search"
                value=""
/>,
              <a
                className="search-form__button m-0 p-0"
                href="#"
                onClick={[Function]}
                role="button"
                type="submit"
>
                <img
                                alt="search icon"
                                src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                width="20"
                />
</a>,
            ],
            "className": "search-form my-2 my-lg-0 col-flex",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-form__input bg-dark",
                "onChange": [Function],
                "placeholder": "Search",
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="search icon"
                  src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                  width="20"
/>,
                "className": "search-form__button m-0 p-0",
                "href": "#",
                "onClick": [Function],
                "role": "button",
                "type": "submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search icon",
                  "src": "https://icongr.am/clarity/search.svg?color=FFFFFF",
                  "width": "20",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="navbar bg-nav row-flex"
>
          <p
                    className="navbar-brand m-0 py-0 pr-2"
                    href="#"
          >
                    <img
                              alt="itunes"
                              className="d-inline-block align-center"
                              height="40"
                              src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                              width="40"
                    />
                    iTunes
          </p>
          <form
                    className="search-form my-2 my-lg-0 col-flex"
                    onSubmit={[Function]}
          >
                    <input
                              className="search-form__input bg-dark"
                              onChange={[Function]}
                              placeholder="Search"
                              type="search"
                              value=""
                    />
                    <a
                              className="search-form__button m-0 p-0"
                              href="#"
                              onClick={[Function]}
                              role="button"
                              type="submit"
                    >
                              <img
                                        alt="search icon"
                                        src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                        width="20"
                              />
                    </a>
          </form>
</div>,
        "className": "justify-content-center align-items-center search-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="navbar-brand m-0 py-0 pr-2"
              href="#"
>
              <img
                            alt="itunes"
                            className="d-inline-block align-center"
                            height="40"
                            src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                            width="40"
              />
              iTunes
</p>,
            <form
              className="search-form my-2 my-lg-0 col-flex"
              onSubmit={[Function]}
>
              <input
                            className="search-form__input bg-dark"
                            onChange={[Function]}
                            placeholder="Search"
                            type="search"
                            value=""
              />
              <a
                            className="search-form__button m-0 p-0"
                            href="#"
                            onClick={[Function]}
                            role="button"
                            type="submit"
              >
                            <img
                                          alt="search icon"
                                          src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                          width="20"
                            />
              </a>
</form>,
          ],
          "className": "navbar bg-nav row-flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="itunes"
                  className="d-inline-block align-center"
                  height="40"
                  src="https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg"
                  width="40"
/>,
                "iTunes",
              ],
              "className": "navbar-brand m-0 py-0 pr-2",
              "href": "#",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "itunes",
                  "className": "d-inline-block align-center",
                  "height": "40",
                  "src": "https://www.apple.com/ac/globalnav/3/fr_FR/images/globalnav/apple/image_large.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "iTunes",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search-form__input bg-dark"
                  onChange={[Function]}
                  placeholder="Search"
                  type="search"
                  value=""
/>,
                <a
                  className="search-form__button m-0 p-0"
                  href="#"
                  onClick={[Function]}
                  role="button"
                  type="submit"
>
                  <img
                                    alt="search icon"
                                    src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                                    width="20"
                  />
</a>,
              ],
              "className": "search-form my-2 my-lg-0 col-flex",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search-form__input bg-dark",
                  "onChange": [Function],
                  "placeholder": "Search",
                  "type": "search",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="search icon"
                    src="https://icongr.am/clarity/search.svg?color=FFFFFF"
                    width="20"
/>,
                  "className": "search-form__button m-0 p-0",
                  "href": "#",
                  "onClick": [Function],
                  "role": "button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search icon",
                    "src": "https://icongr.am/clarity/search.svg?color=FFFFFF",
                    "width": "20",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
